core/store/src/migrations.rs

```diff
@@ -186,3 +200,54 @@ pub fn migrate_31_to_32(storage: &crate::NodeStorage) -> anyhow::Result<()> {
     update.commit()?;
     Ok(())
 }
+
+fn get_row_count(store: &Store, col_name: &str) -> Option<i64> {
+    let statistics = store.get_store_statistics()?;
+    let col_stats = statistics.data.into_iter().find(|(col, _)| col == col_name)?.1;
+    for stat in col_stats {
+        match stat {
+            crate::db::StatsValue::Count(count) => return Some(count),
+            _ => {}
+        }
+    }
+    None
+}
+
+/// Migrates database from version 32 to 33.
+///
+/// This removes the TransactionResult column and moves it to TransactionResultForBlock.
+/// The new column removes the need for high-latency read-modify-write operations when committing
+/// new blocks.
+pub fn migrate_32_to_33(storage: &crate::NodeStorage) -> anyhow::Result<()> {
```

Comment:
Have you run it on RPC and archival nodes so that we know what’s the approximate time for this?

Reply:
Not yet, that's what I'm trying next :D

Reply:
On a RPC node the migration took around 1m50s; old column was 2.8GB, new column was 2.7GB.
On an archival node the migration took 2hr14min; old column was 250GB, new column was 251GB.